src/main/java/ulcambridge/foundations/viewer/DocumentViewController.java

```diff
@@ -319,6 +331,84 @@ private String getCanonicalItemUrl(String itemId, int page) {
                 .toUriString();
     }
 
+    /**
+     * Get the IIIF URL for a social media link preview image at a specific page.
+     *
+     * @param json The item JSON.
+     * @param page The page of the item to get the image URL for.
+     * @param imgDims The dimensions (width, height) of image to request.
+     * @return The IIIF URL
+     */
+    private String getSocialIIIFUrl(JSONObject json, int page, Map<String, String> imgDims) {
```

Comment:
We could separate this out into a separate class, but I don't think it's a big issue to have it within the DocumentViewController.  We may want to refactor it in future especially if we look at linking collections in the future and generating social media images for that.

Reply:
If you want to move it, we could make it a separate class for SocialMediaImages within the util package.  I don't think it's essential but if it helps with testing, that might be good.